src/Integration.Vsix/DaemonIntegrationPackage.cs

```diff
@@ -0,0 +1,172 @@
+    public sealed class DaemonIntegrationPackage : Package
+    {
+        private static readonly string DAEMON_HOST = "localhost";
+        private static readonly int DAEMON_PORT = 8050;
+
+        /// <summary>
+        /// SonarLintDaemonIntegrationPackage GUID string.
+        /// </summary>
+        public const string PackageGuidString = "6f63ab5a-5ab8-4a0d-9914-151911885966";
+
+        internal static DaemonIntegrationPackage Instance { get; private set; }
+
+        private string WorkDir;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DaemonIntegrationPackage"/> class.
+        /// </summary>
+        public DaemonIntegrationPackage()
+        {
+            // Inside this method you can place any initialization code that does not require
+            // any Visual Studio service because at this point the package object is created but
+            // not sited yet inside Visual Studio environment. The place to do all the other
+            // initialization is the Initialize method.
+        }
+
+        #region Package Members
+
+        /// <summary>
+        /// Initialization of the package; this method is called right after the package is sited, so this is the place
+        /// where you can put all the initialization code that rely on services provided by VisualStudio.
+        /// </summary>
+        protected override void Initialize()
+        {
+            base.Initialize();
+
+            InitializeSonarLintDaemonFacade();
+
+            Instance = this;
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+            if (disposing)
+            {
+                Directory.Delete(WorkDir, true);
+            }
+        }
+
+        #endregion
+
+        private void InitializeSonarLintDaemonFacade()
+        {
+            WorkDir = CreateTempDir();
+        }
+
+        private string CreateTempDir()
+        {
+            string tempDirectory = Path.Combine(Path.GetTempPath(), "SonarLintDaemon", Path.GetRandomFileName());
+            Directory.CreateDirectory(tempDirectory);
+            return tempDirectory;
+        }
+
+        public void RequestAnalysis(string path, string charset)
+        {
+            Analyze(path, charset);
+        }
+
+        private async void Analyze(string path, string charset)
+        {
+            var channel = new Channel(string.Join(":", DAEMON_HOST, DAEMON_PORT), ChannelCredentials.Insecure);
+            var client = new StandaloneSonarLint.StandaloneSonarLintClient(channel);
+
+            var inputFile = new InputFile
+            {
+                Path = path,
+                Charset = charset,
+            };
+
+            var request = new AnalysisReq
+            {
+                // TODO get correct basedir
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)